app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddCyclewayPartSurface.kt

```diff
@@ -48,6 +48,7 @@ class AddCyclewayPartSurface : OsmFilterQuestType<SurfaceAnswer>() {
                 changes.updateWithCheckDate("cycleway:surface", answer.value.osmValue)
                 changes.addOrModify("cycleway:surface:note", answer.note)
             }
+            IsActuallyStepsSurfaceAnswer -> throw IllegalStateException()
```

Comment:
Sealed classes exist to avoid these kinds of things

```kotlin
sealed class SurfaceOrIsStepsAnswer
object IsActuallyStepsAnswer : SurfaceOrIsStepsAnswer()

sealed class SurfaceAnswer : SurfaceOrIsStepsAnswer()
data class GenericSurfaceAnswer(val value: Surface, val note: String) : SurfaceAnswer()
data class SpecificSurfaceAnswer(val value: Surface) : SurfaceAnswer()

```

Reply:
Fixed in [`6715580` (#3334)](https://github.com/streetcomplete/StreetComplete/pull/3334/commits/6715580eb934a694c922d2e24c2a0d952f0c2343)